### PR TITLE
Add template document for Azure PowerShell releases

### PR DESCRIPTION
**Pull request by cormacpayne**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@cormacpayne__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (corm). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: documentation/azure-powershell-release-template.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    - For example, for the August 2016 release of Azure PowerShell 2.0.0, the command would look like the following:
+    > `.\PreparePSRelease.ps1 2.0.0 "August 2016" -Major`
+- [ ] Create a signed build from the release branch using the **powershell-sign** Jenkins job
+- [ ] Once the build is successful, click on *Build Artifacts* and download the zip
+    - This zip file will contain the .msi for Azure PowerShell, and all of its packages
+- [ ] Install Orca from the .msi in the following folder: **\\\\aaptfile01\adxsdk\PowerShell**
+- [ ] In the zip file, rename the .msi to **azure-powershell.X.X.X.msi**
+- [ ] Open this .msi in Orca, select the *Property* page in the left panel, and save the *ProductCode* somewhere you can easily access it
+- [ ]  Create an account on [CodePlex](codeplex.com) if you don't already have one, and link it with your Microsoft corporate account
+- [ ]  To access the [Web Platform Install Feeds](https://webpi.codeplex.com/) repository,  contact [Chris Sfanos](mailto:chris.sfanos@microsoft.com) with your CodePlex username and request to join the Web Platform Installer project
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: I would put this preparatory step up higher, and caveat it that it needs to be done one time only,

File: documentation/azure-powershell-release-template.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    - You can check the settings for the **dev** branch if you have any questions
+- [ ] Clone the Azure PowerShell repo to your computer, and switch to the release branch
+- [ ] Open a PowerShell window, and run the script **tools\PreparePSRelease.ps1**
+    - Check the script header for a sample of how the parameters should look
+    - For example, for the August 2016 release of Azure PowerShell 2.0.0, the command would look like the following:
+    > `.\PreparePSRelease.ps1 2.0.0 "August 2016" -Major`
+- [ ] Create a signed build from the release branch using the **powershell-sign** Jenkins job
+- [ ] Once the build is successful, click on *Build Artifacts* and download the zip
+    - This zip file will contain the .msi for Azure PowerShell, and all of its packages
+- [ ] Install Orca from the .msi in the following folder: **\\\\aaptfile01\adxsdk\PowerShell**
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: I don't think we can put the share name in

File: documentation/azure-powershell-release-template.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    - Make sure to tell the partners to update the change log with any changes that they made that aren't currently found in it for the release
+    - Copy the e-mail contents and post it on the slack channels (specifically [#powershell](https://azuresdk.slack.com/messages/powershell/) and [#powershell-announce](https://azuresdk.slack.com/messages/powershell-announce/))
+- [ ] Give anyone with an open PR, or those that are making quick additions, a chance to create a new PR that targets the release branch, get their code reviewed, and run a successful **powershell-demand** job
+- [ ] Run the **powershell-sign** job on the release branch and notify partner teams of this release candidate
+    - **Note:** For additional release candidates created, make sure to update the **DH_AzurePS.xml** file with the new product code of the .msi (obtained from Orca)
+- [ ] Wait for partner teams to sign off by MM/DD/YYYY (a few days after the initial release candidate was sent out)
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: a few days == 3 days

File: documentation/azure-powershell-release-template.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- [ ] Make sure that you have registered a PSRepository for the publish location
+    - To see if you have a PSRepository for this location, run the following command: `Get-PSRepository`; if any object with the `SourceLocation` of http://dtlgalleryint.cloudapp.net/api/v2 is listed, you already have the correct PSRepository
+    - If this PSRepository does not show up, run the following command to register it: `Register-PSRepository -Name dtgallery -SourceLocation http://dtlgalleryint.cloudapp.net/api/v2 -PublishLocation http://dtlgalleryint.cloudapp.net/api/v2 -InstallationPolicy Trusted`
+- [ ] On the evening of MM/DD/YYYY, publish the latest RC of Azure PowerShell to internal through the **ps-pub-test** Jenkins job and smoke test from [here](dtlgalleryint.cloudapp.net/api/v2)
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: On the evening before the release

File: documentation/azure-powershell-release-template.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    - Install the modules for testing by running the following commands: `Install-Module -Repository dtlgallery -Name AzureRM` and `Install-Module -Repository dtlgallery -Name Azure -AllowClobber`
+    - Run the **RunInstallationTests.ps1** script again (same one from the test signoff document) and ensure that all tests passed
+- [ ] Install [Azure Storage Explorer](https://azurestorageexplorer.codeplex.com/) and upload the .msi file to the download blob container
+    - The storage account name and storage account key can be retrieved from the OneNote page
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: From our KeyVault, do not mention the key name

File: documentation/azure-powershell-release-template.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    - Use the change log from this release as the content of the document
+    - Include the .msi file and include a link to it at the top of the document with "Installer: Windows Standalone"
+    - Include the changes since the last release at the end of the document
+- [ ] Merge the release branch to master
+- [ ] Delete the release branch
+- [ ] Merge the master branch to dev
+- [ ] Smoke test the live WebPI feed
+- [ ] Send an email to [all partners](mailto:adxsdkpartners@microsoft.com) with the locations of where the release can be accessed: the PowerShell Gallery modules (both Azure and AzureRM), the WebPI feed, and the GitHub release
+    - Copy the e-mail contents and post it on the slack channels (specifically [#powershell](https://azuresdk.slack.com/messages/powershell/) and [#powershell-announce](https://azuresdk.slack.com/messages/powershell-announce/))
+- [ ] Resolve any issues, close the release milestone, and create a new milestone for +3 months
```
Comment (markcowl) [inline documentation/azure-powershell-release-template.md]: Contact partners about any of their bugs still open in the release, and ensure bugs are closed or moved into the appropriate milestone

**Pull request closed without merging**